clusterloader2/pkg/measurement/common/network/performance-measurement-helper.go

```diff
@@ -0,0 +1,202 @@
+	}
+	return true, nil
+}
+
+func deriveClientServerPodNum(ratio string) (int, int, string) {
+	var podNumber []string
+	var clientPodNum, serverPodNum int
+
+	podNumber = strings.Split(ratio, ratioSeparator)
+	clientPodNum, _ = strconv.Atoi(podNumber[0])
+	serverPodNum, _ = strconv.Atoi(podNumber[1])
+	if clientPodNum == serverPodNum && clientPodNum == 1 {
+		return clientPodNum, serverPodNum, oneToOne
+	}
+	if (clientPodNum > serverPodNum) && serverPodNum == 1 {
+		return clientPodNum, serverPodNum, manyToOne
+	}
+	if clientPodNum == serverPodNum {
+		return clientPodNum, serverPodNum, manyToMany
+	}
+	return clientPodNum, serverPodNum, invalid
+}
+
+func (npm *networkPerformanceMeasurement) populateTemplate(podPair uniquePodPair, index int) map[string]interface{} {
+	return map[string]interface{}{
+		"index":                index,
+		"clientPodName":        podPair.sourcePodName,
+		"serverPodName":        podPair.destinationPodName,
+		"clientPodIP":          podPair.sourcePodIP,
+		"serverPodIP":          podPair.destinationPodIP,
+		"protocol":             npm.protocol,
+		"duration":             npm.testDuration,
+		"clientStartTimestamp": npm.startTimeStampForTestExecution,
+		"numberOfClients":      1}
+}
+
+func getUnusedPod(unusedPodList *[]workerPodData) (workerPodData, error) {
+	var unusedPod workerPodData
+	if len(*unusedPodList) == 0 {
+		return unusedPod, errors.New("unused pod list empty")
+	}
+	numOfPods := len(*unusedPodList)
+	// extract last pod of slice
+	unusedPod = (*unusedPodList)[numOfPods-1]
+	*unusedPodList = (*unusedPodList)[:numOfPods-1]
+	return unusedPod, nil
+}
+
+func getStartTimeStampForPod() int64 {
+	currTime := time.Now()
+	initDelay := time.Second * time.Duration(initialDelayForTestExecution)
+	futureTime := currTime.Add(initDelay).Unix()
+	return futureTime
+}
+
+func getUnit(metric string) string {
+	return metricUnitMap[metric]
+}
+
+func (npm *networkPerformanceMeasurement) closeCh() {
+	close(npm.podPairCh)
+}
+
+type float64Slice []float64
+
+func (p float64Slice) Len() int           { return len(p) }
+func (p float64Slice) Less(i, j int) bool { return p[i] < p[j] }
+func (p float64Slice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+func getPercentile(values float64Slice, percentile float64) float64 {
+	ps := []float64{percentile}
+
+	scores := make([]float64, len(ps))
+	size := len(values)
+	if size <= 0 {
+		return scores[0]
+	}
+
+	sort.Sort(values)
+	for i, p := range ps {
+		pos := p * float64(size+1)
+		if pos < 1.0 {
+			scores[i] = values[0]
+		} else if pos >= float64(size) {
+			scores[i] = values[size-1]
+		} else {
+			lower := values[int(pos)-1]
+			upper := values[int(pos)]
+			scores[i] = lower + (pos-math.Floor(pos))*(upper-lower)
+		}
+	}
+	return scores[0]
+}
+
+func getMetricResponse(metricMap map[string]interface{}, response *metricResponse) {
+	var body []byte
+	body, err := json.Marshal(metricMap)
+	if err != nil {
+		klog.Error("error marshaling metricMap:", err)
```

Comment:
if we error here we can return, the Unmarshall will fail later, right?

Reply:
Yes its handled now.